File: api/bookings.js
import { supabase } from './utils/supabaseClient.js';

export default async function handler(req, res) {
  // Set CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (!supabase) {
    console.error('Supabase is not configured for bookings handler');
    return res.status(500).json({
      success: false,
      error: 'Database not configured. Please set SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY.',
    });
  }

  try {
    if (req.method === 'GET') {
      // Stats mode: /api/bookings?stats=1
      if (req.query.stats) {
        const { data, error } = await supabase
          .from('bookings')
          .select('status,payment_status,price');

        if (error) {
          console.error('Error fetching booking stats:', error);
          return res.status(500).json({ success: false, error: 'Failed to load booking stats' });
        }

        const totalBookings = data.length;
        const confirmed = data.filter((b) => b.status === 'confirmed').length;
        const pending = data.filter((b) => b.status === 'pending').length;
        const revenue = data
          .filter((b) => b.payment_status === 'paid')
          .reduce((sum, b) => sum + (b.price || 0), 0);

        return res.status(200).json({
          success: true,
          data: { totalBookings, confirmed, pending, revenue },
        });
      }

      // List bookings
      const { data, error } = await supabase
        .from('bookings')
        .select('*')
        .order('created_at', { ascending: false });

      if (error) {
        console.error('Error fetching bookings:', error);
        return res.status(500).json({ success: false, error: 'Failed to load bookings' });
      }

      return res.status(200).json({
        success: true,
        data: data || [],
      });
    }

    if (req.method === 'POST') {
      const payload = {
        ...req.body,
      };

      const { data, error } = await supabase
        .from('bookings')
        .insert(payload)
        .select('*')
        .single();

      if (error) {
        console.error('Error creating booking:', error);
        return res.status(500).json({ success: false, error: 'Failed to create booking' });
      }

      return res.status(201).json({
        success: true,
        data,
      });
    }

    if (req.method === 'PUT') {
      const bookingId = req.query.id;
      if (!bookingId) {
        return res.status(400).json({ success: false, error: 'Missing booking id' });
      }

      const { data, error } = await supabase
        .from('bookings')
        .update(req.body)
        .eq('id', bookingId)
        .select('*')
        .single();

      if (error) {
        console.error('Error updating booking:', error);
        return res.status(500).json({ success: false, error: 'Failed to update booking' });
      }

      return res.status(200).json({
        success: true,
        data,
      });
    }

    if (req.method === 'DELETE') {
      const bookingId = req.query.id;
      if (!bookingId) {
        return res.status(400).json({ success: false, error: 'Missing booking id' });
      }

      const { error } = await supabase.from('bookings').delete().eq('id', bookingId);

      if (error) {
        console.error('Error deleting booking:', error);
        return res.status(500).json({ success: false, error: 'Failed to delete booking' });
      }

      return res.status(200).json({
        success: true,
        message: 'Booking deleted successfully',
      });
    }

    res.setHeader('Allow', ['GET', 'POST', 'PUT', 'DELETE']);
    return res.status(405).end(`Method ${req.method} Not Allowed`);
  } catch (err) {
    console.error('Unexpected error in bookings handler:', err);
    return res.status(500).json({ success: false, error: 'Internal server error' });
  }
}

File: api/codefiles/bookings.py
import json
from http.server import BaseHTTPRequestHandler

class BookingsHandler(BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type, Authorization')
        self.end_headers()

    def do_GET(self):
        # Demo data for bookings
        demo_bookings = [
            {
                'id': '1',
                'customerName': 'John Doe',
                'customerEmail': 'john@example.com',
                'customerPhone': '9876543210',
                'service': 'Tarot Reading',
                'date': '2024-02-15',
                'time': '10:00 AM',
                'status': 'confirmed',
                'paymentStatus': 'paid',
                'amount': 1500,
                'createdAt': '2024-02-10T10:00:00Z'
            },
            {
                'id': '2',
                'customerName': 'Jane Smith',
                'customerEmail': 'jane@example.com',
                'customerPhone': '9876543211',
                'service': 'Love Reading',
                'date': '2024-02-16',
                'time': '2:00 PM',
                'status': 'pending',
                'paymentStatus': 'pending',
                'amount': 1200,
                'createdAt': '2024-02-11T14:00:00Z'
            }
        ]
        
        # Demo stats
        demo_stats = {
            'totalBookings': len(demo_bookings),
            'confirmed': len([b for b in demo_bookings if b['status'] == 'confirmed']),
            'pending': len([b for b in demo_bookings if b['status'] == 'pending']),
            'revenue': sum([b['amount'] for b in demo_bookings if b['paymentStatus'] == 'paid'])
        }
        
        # Check if requesting stats
        if self.path.endswith('/stats'):
            response = {'success': True, 'data': demo_stats}
        else:
            response = {'success': True, 'data': demo_bookings}
        
        self.send_response(200)
        self.send_header('Content-Type', 'application/json')
        self.end_headers()
        self.wfile.write(json.dumps(response).encode('utf-8'))

    def do_POST(self):
        # Create new booking
        content_length = int(self.headers.get('Content-Length', 0))
        post_data = self.rfile.read(content_length)
        
        try:
            data = json.loads(post_data.decode('utf-8'))
            new_booking = {
                'id': str(len(demo_bookings) + 1),
                **data,
                'status': 'pending',
                'paymentStatus': 'pending',
                'createdAt': f'{__import__("datetime").datetime.now().isoformat()}Z'
            }
            
            response = {'success': True, 'data': new_booking}
            
        except Exception as e:
            response = {'success': False, 'error': str(e)}
        
        self.send_response(201)
        self.send_header('Content-Type', 'application/json')
        self.end_headers()
        self.wfile.write(json.dumps(response).encode('utf-8'))

def handler(request):
    return BookingsHandler()(request)
